src/renderers/dom/client/syntheticEvents/SyntheticEvent.js

```diff
@@ -90,17 +100,21 @@ function SyntheticEvent(dispatchConfig, targetInst, nativeEvent, nativeEventTarg
 assign(SyntheticEvent.prototype, {
 
   preventDefault: function() {
+    if (this._destructored) {
+      if (__DEV__) {
+        warning(
+          event,
+          'This synthetic event is reused for performance reasons. If you\'re ' +
+          'seeing this, you\'re calling `preventDefault` on a ' +
+          'released/nullified synthetic event. This is a no-op. See ' +
+          'https://fb.me/react-event-pooling for more information.'
+        );
+      }
+      return;
+    }
+
     this.defaultPrevented = true;
     var event = this.nativeEvent;
-    if (__DEV__) {
```

Comment:
I think you can do the opposite:

* Remove the warnings from method implementations
* In the destructor, use `Object.defineProperty` to put functions-that-just-emit-warnings (and noop in production) instead of the real implementations